feat: Implement new search components

firestack opened the pull request:
This does a few things to create new components based on old components, while avoiding accidentally modifying the old components.
- Copy `SearchForm` to `OldSearchForm`, update references to point to `OldSearchForm`
- Update `SearchForm` to match new designs, use on Map Page.
- Create `FilterAccordion` mostly from `GarageFilter`
	- It's not _exactly_ the same, so I didn't feel comfortable making the `GarageFilter` depend on this component. But I do think we could. Will either implement here or open a ticket based on dev feedback.
- Build a "interactive"(focus state, similar to the bus stop icons) `CircleXIcon`
	- note: I don't like that this component name matches that of our `CircleXIcon` from our `icons.tsx` file, but I was unsure what to name it otherwise.

Additionally, the Filter currently acts like the old ratio buttons. Because the application reducer that remembers these settings needs to change to accommodate this, I've left this to be changed in another PR. If Design pushes back on this, I'll update this PR with those changes.

---

Asana Ticket: https://app.asana.com/0/0/1204650425301364

lemald commented on assets/css/_search_form.scss:
```diff
@@ -1,95 +1,97 @@
   position: relative;
-  flex: 1 1 auto;
+
+  // Use bootstrap control as base for now
+  @extend .form-control;
```
**question:** what are the relative merits of extending the `form-control` style versus customizing it via the available Sass variables? It seems like some of this styling is definitely more specific (mainly leaving that extra space for the controls), but other parts about the colors and focus state look like they could maybe be accommodated via the Sass method.

firestack replied:
> versus customizing it via the available Sass variables
I didn't want to assume we'd be going with bootstrap as a design system, so I didn't want to spend time customizing bootstrap, and instead try and be as close to the way we're currently doing CSS within Skate.

My thought process for extending `form-control` was
- it should give us an easier transistion to bootstrap for this component if we use bootstrap as our design system.
- it should allow us to reference an existing component+style if we're moving to a different design system.

I contemplated inlining bootstraps styles or pruning it down, but I figured the dev time wasn't worth it since this worked.

I'm open to configuring this via SASS/CSS variables if you think it's worth it, I just didn't want to assume that this is what _all_ our `form-controls` should follow in terms of color/focus.

lemald replied:
Makes sense! I wasn't sure exactly how all of your conversations with design had panned out, and I understand not wanting to make premature decisions about what parts of this styling will be universal.

lemald commented on assets/css/_search_form.scss:
```diff
@@ -1,95 +1,97 @@
+// #region Search Input Buttons
+.c-search-form__input-controls {
+  position: absolute;
+  // Stick to right side of box, do not intercept mouse events to the text input
+  inset-block: 0;
+  right: 0;
```
**question:** what's the purpose of setting both `inset-block` and `right`? I assume there's some reason but I can't seem to figure it out.

firestack replied:
I needed to set everything _but_ `left`, here, `inset-block` sets both `top` and `bottom` to `0`.

Alternatively I think I could do `inset: 0` and `left: unset`? Unsure about this though (I thought I tried it).
```suggestion
  // Stick to right side of box, do not intercept mouse events to the text input
  inset: 0;
  left: 0;
```

lemald replied:
Ah, okay, I was slightly misreading the MDN docs on `inset-block` and that's not a property I was previously familiar with. I think it's fine as-is.

lemald replied:
On the other hand maybe it's weird to mix `left` / `right` / `top` / `bottom` with properties that depend on text directionality and writing mode stuff, but I guess the practical chance of that being an issue is extremely low.

firestack replied:
True! I'm gonna rework this

firestack replied:
[`f812089` (#2128)](https://github.com/mbta/skate/pull/2128/commits/f81208973c7f11e338010fdb826601338d164377)

lemald commented on assets/css/_search_form.scss:
```diff
@@ -1,95 +1,97 @@
-  list-style-type: none;
-  margin: 0;
-}
+  // flex: 0 0 2.25rem;
+  // flex: 0;
```
**nitpick:** looks like this might be some temporary code that got left over as comments.

lemald reviewed (approved):
Okay, I left a few questions but I don't think any of them are major blockers - overall this looks good!

firestack commented:
@mbta/skate-developers I've gotten final approval from design but there have been a few code changes since last review, I'll plan to merge this and #2132 around 1550-1600 EST if there are no further comments.
